### ISSUE TITLE
Multiple annotations in the same event break mission parsing

### ISSUE THREAD

**Issue by naomimyselfandi** [scp-fs2open/fs2open.github.com#3267]
With the new FRED annotations feature, it's possible to save a mission that won't parse correctly. With really bad luck, it's possible to save a mission in a state that FRED can't even open it.

There are two reliable ways to reproduce this bug:

1. Create an event (a generic `( when ( true ) ( do-nothing ) )` will suffice). Annotate one of the operators, then annotate the event name.
2. Create an event and annotate three of its nodes.

----

This parses correctly:

```
$Formula: ( when ( false ) ( do-nothing ) )
+Name: Continue campaign
+Repeat Count: 1
+Interval: 1
$Annotations Start
+Comment: TODO: This comes true when the player passes the mission.
$end_multi_text
+Background Color: 255, 0, 0
$Annotations End
```

So does this:

```
$Formula: ( when ( false ) ( do-nothing ) )
+Name: Continue campaign
+Repeat Count: 1
+Interval: 1
$Annotations Start
+Comment: TODO: This comes true when the player passes the mission.
$end_multi_text
+Background Color: 255, 0, 0
+Path: 0, 0
$Annotations End
```

This, however, does not (pay close attention to the placement of the `+Path:`), and it results from case 1 up above.

```
$Formula: ( when ( false ) ( do-nothing ) )
+Name: Continue campaign
+Repeat Count: 1
+Interval: 1
$Annotations Start
+Comment: TODO: This comes true when the player passes the mission.
$end_multi_text
+Path: 0, 0
+Background Color: 255, 0, 0
$Annotations End
```

Here's the error:

```
Error: C:\Users\PC\AppData\Local\Knossos\library\FS2\BtA-2.0.0\bta_core\data\missions\0_██████Template.fs2(line 617):
Error: Missing required token: [$Annotations End]. Found [Background Color: 255, 0, 0] instead.

File: parselo.cpp
Line: 309

ntdll.dll! NtWaitForSingleObject + 20 bytes
KERNELBASE.dll! WaitForSingleObjectEx + 147 bytes
fred2_open_21_1_0_20210301_66b7b011c_x64_SSE2.exe! <no symbol>
```

This is on FSO 21.1.0-20210301; I've only tested it in FRED.

Also, the error message really does say `Background Color:`, not `+Background Color:`.

**Comment by Goober5000** [comment]
This is a very good and detailed report, thank you.  I'll take a look.

**Comment by Goober5000** [comment]
I am not able to get FRED to save the +Path out of order like that.  Did you get this result by editing the mission in FRED, or did you edit it in a text editor?  +Path is intentionally designed to be the last element in an annotation, so it should never appear in the middle of two other elements.

If you got the result by using FRED, do you have a sequence of steps that will reproduce this?

**Comment by Goober5000** [comment]
That being said, this detail is important:

> Also, the error message really does say `Background Color:`, not `+Background Color:`.

You've identified a separate bug that may have some bearing on this bug.  That's definitely something I can fix.

**Comment by naomimyselfandi** [comment]
> I am not able to get FRED to save the +Path out of order like that. Did you get this result by editing the mission in FRED, or did you edit it in a text editor? +Path is intentionally designed to be the last element in an annotation, so it should never appear in the middle of two other elements.
> 
> If you got the result by using FRED, do you have a sequence of steps that will reproduce this?

I did manually edit the file so I could get it to open in FRED, so it's possible I messed up while submitting the issue, but I don't think I did.

I don't think it's a matter of "out of order" so much as "multiple entries". To get the third (broken) case, I just applied a comment to the event name, and a background color to a node in the event. (At that point, I was trying to narrow down the issue.) Does that make sense?

If that doesn't clear it up, I can double-check in the morning.

**Comment by naomimyselfandi** [comment]
Okay, now I'm beyond confused. I tried to repro this on the same version of FRED, with the same mission, and the mission happily saved and opened back up without any fuss. Earlier today it was as easy as applying multiple annotations to (multiple nodes in) the same event. (And I only started editing the mission in vim _because_ of this bug, so that's not it!)

I'm just spitballing here, but maybe there's some underlying parser bug that's causing both this issue and the mysterious vanishing plus sign?

**Comment by naomimyselfandi** [comment]
_Ah!_

Okay, I'm sorry for Battuta-posting, but I may have just figured it out. Try this: Create an event, apply an annotation to some node in the event, then apply an annotation to a node to the event name or a node that comes earlier. (So with the generic `( when ( true ) ( do-nothing ) )` event, you could annotate the `do-nothing`, then the event name or the `true`. _That_ seems to reliably trigger it!

**Comment by Goober5000** [comment]
This is fixed by #3269.

**Comment by Goober5000** [comment]
Fix merged.
